torch/utils/data/datapipes/utils/common.py

```diff
@@ -20,6 +21,88 @@
 ]
 
 
+def validate_input_col(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a callable datapipe works with the input column
+
+    This simply ensures that the number of
+     positional arguments matches
+    the size of the input column. The function must not contain any non-default
+    keyword-only arguments.
+
+    Examples:
+        >>> def f(a, b, *, c=1):
+        >>>     return a + b + c
+        >>> def f_def(a, b=1, *, c=1):
+        >>>     return a + b + c
+        >>> assert validate_input_col(f, [1, 2])
+        >>> assert validate_input_col(f_def, 1)
+        >>> assert validate_input_col(f_def, [1, 2])
+
+    Notes:
+        If the function contains variable positional (`inspect.VAR_POSITIONAL`) arguments,
+        for example, f(a, *args), the validator will accept any size of input column
+        greater than or equal to the number of positional or keyword arguments.
```

Comment:
nit: `greater than or equal to the number of positional.`

We don't take keyword argument